sonar-scanner-engine/src/main/java/org/sonar/scanner/genericcoverage/GenericTestExecutionReportParser.java

```diff
@@ -0,0 +1,159 @@
+    new StaxParser(rootCursor -> {
+      rootCursor.advance();
+      parseRootNode(rootCursor, context);
+    }).parse(inputStream);
+  }
+
+  private void parseRootNode(SMHierarchicCursor rootCursor, SensorContext context) throws XMLStreamException {
+    checkElementName(rootCursor, "unitTest");
+    String version = rootCursor.getAttrValue("version");
+    if (!"1".equals(version)) {
+      throw new IllegalStateException("Unknown report version: " + version + ". This parser only handles version 1.");
+    }
+    parseFiles(rootCursor.childElementCursor(), context);
+  }
+
+  private void parseFiles(SMInputCursor fileCursor, SensorContext context) throws XMLStreamException {
+    while (fileCursor.getNext() != null) {
+      checkElementName(fileCursor, "file");
+      String filePath = mandatoryAttribute(fileCursor, "path");
+      InputFile inputFile = context.fileSystem().inputFile(context.fileSystem().predicates().hasPath(filePath));
+      if (inputFile == null) {
+        numberOfUnknownFiles++;
+        if (numberOfUnknownFiles <= MAX_STORED_UNKNOWN_FILE_PATHS) {
+          firstUnknownFiles.add(filePath);
+        }
+        continue;
+      }
+      Preconditions.checkState(
+        inputFile.language() != null,
+        "Line %s of report refers to a file with an unknown language: %s",
+        fileCursor.getCursorLocation().getLineNumber(),
+        filePath);
+      Preconditions.checkState(
+        inputFile.type() != InputFile.Type.MAIN,
+        "Line %s of report refers to a file which is not configured as a test file: %s",
+        fileCursor.getCursorLocation().getLineNumber(),
+        filePath);
+      matchedFileKeys.add(inputFile.absolutePath());
+
+      MutableTestPlan testPlan = testPlanBuilder.loadPerspective(MutableTestPlan.class, inputFile);
+      SMInputCursor testCaseCursor = fileCursor.childElementCursor();
+      while (testCaseCursor.getNext() != null) {
+        parseTestCase(testCaseCursor, testPlan);
+      }
+    }
+  }
+
+  private void parseTestCase(SMInputCursor cursor, MutableTestPlan testPlan) throws XMLStreamException {
+    checkElementName(cursor, "testCase");
+    MutableTestCase testCase = testPlan.addTestCase(mandatoryAttribute(cursor, NAME_ATTR));
+    TestCase.Status status = TestCase.Status.OK;
+    testCase.setDurationInMs(longValue(mandatoryAttribute(cursor, DURATION_ATTR), cursor, DURATION_ATTR, 0));
+
+    SMInputCursor child = cursor.descendantElementCursor();
+    if (child.getNext() != null) {
+      String elementName = child.getLocalName();
+      if (SKIPPED.equals(elementName)) {
+        status = TestCase.Status.SKIPPED;
+      } else if (FAILURE.equals(elementName)) {
+        status = TestCase.Status.FAILURE;
+      } else if (ERROR.equals(elementName)) {
+        status = TestCase.Status.ERROR;
+      }
+      testCase.setStatus(status);
+      if (TestCase.Status.OK != status) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "TestCase"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)